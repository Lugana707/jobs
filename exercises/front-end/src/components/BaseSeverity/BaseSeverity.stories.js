import '~/assets/scss/styles.scss';
import BaseSeverity from './BaseSeverity';

export default {
  title: 'Library/Base Components/Severity',
  component: BaseSeverity,
};

export const Default = () => ({
  components: { BaseSeverity },
  props: {
    issuesCritical: {
      default: 1,
    },
    issuesHigh: {
      default: 0,
    },
    issuesMedium: {
      default: 0,
    },
    issuesLow: {
      default: 0,
    },
    licensesHigh: {
      default: 0,
    },
    licensesMedium: {
      default: 0,
    },
    licensesLow: {
      default: 0,
    },
    size: {
      default: null,
    },
    compact: {
      default: false,
    },
    verbose: {
      default: false,
    },
  },
  template: `
  <BaseSeverity 
    :issues-high="issuesHigh"
    :licenses-high="licensesHigh"
    :issues-medium="issuesMedium"
    :licenses-medium="licensesMedium"
    :issues-low="issuesLow"
    :licenses-low="licensesLow"
    :size="size"
    :compact="compact"
    :verbose="verbose"
  />`,
});

export const Expanded = () => `
  <BaseSeverity
    :issues-high="23"
    :licenses-high="2"
    :issues-medium="6"
    :licenses-medium="1"
    :issues-low="1"
    :licenses-low="0"
  />
`;

export const Compact = () => `
  <BaseSeverity
    :issues-high="23"
    :licenses-high="2"
    :issues-medium="6"
    :licenses-medium="1"
    :issues-low="1"
    :licenses-low="0"
    compact
  />
`;

export const Verbose = () => `
  <BaseSeverity
    :issues-high="4"
    :licenses-high="2"
    :issues-medium="0"
    :licenses-medium="1"
    :issues-low="8"
    :licenses-low="0"
    compact
    verbose
  />
`;

export const Sizes = () => `
  <div>
    <div style="margin-bottom: 24px;">
      <BaseSeverity
        style="margin-right: 24px;"
        :issues-high="23"
        :licenses-high="2"
        :issues-medium="6"
        :licenses-medium="1"
        :issues-low="8"
        :licenses-low="0"
      />
      <BaseSeverity
        :issues-high="23"
        :licenses-high="2"
        :issues-medium="6"
        :licenses-medium="1"
        :issues-low="8"
        :licenses-low="0"
        size="small"
      />
    </div>
    <div>
      <BaseSeverity
        style="margin-right: 24px;"
        :issues-high="23"
        :licenses-high="2"
        :issues-medium="6"
        :licenses-medium="1"
        :issues-low="1"
        :licenses-low="0"
        compact
      />
      <BaseSeverity
        :issues-high="23"
        :licenses-high="2"
        :issues-medium="6"
        :licenses-medium="1"
        :issues-low="1"
        :licenses-low="0"
        size="small"
        compact
      />
    </div>
  </div>
`;

export const Extreme = () => `
  <BaseSeverity
    :issues-high="1876"
    :issues-medium="237654"
    :issues-low="6784923"
  />
`;

export const Critical = () => ({
  components: { BaseSeverity },
  props: {
    showCriticalSeverity: {
      default: true,
    },
  },
  template: `
  <div>
    <div style="margin-bottom: 24px;">
      <BaseSeverity
        style="margin-right: 24px;"
        :issues-critical="5"
        :issues-high="23"
        :licenses-high="2"
        :issues-medium="6"
        :licenses-medium="1"
        :issues-low="8"
        :licenses-low="0"
        :show-critical-severity="showCriticalSeverity"
      />
      <BaseSeverity
      :issues-critical="7"
      :issues-high="23"
      :licenses-high="2"
      :issues-medium="6"
      :licenses-medium="1"
      :issues-low="8"
      :licenses-low="0"
      size="small"
      :show-critical-severity="showCriticalSeverity"
      />
      </div>
      <div style="margin-bottom: 24px;">
      <BaseSeverity
      :issues-critical="0"
      :issues-high="23"
      :licenses-high="2"
      :issues-medium="6"
      :licenses-medium="1"
      :issues-low="8"
      :licenses-low="0"
      size="small"
      :show-critical-severity="showCriticalSeverity"
      />
    </div>
    <div style="margin-bottom: 24px;">
      <BaseSeverity
        style="margin-right: 24px;"
        :issues-critical="9"
        :issues-high="23"
        :licenses-high="2"
        :issues-medium="6"
        :licenses-medium="1"
        :issues-low="1"
        :licenses-low="0"
        compact
        :show-critical-severity="showCriticalSeverity"
      />
      <BaseSeverity
      :issues-critical="2"
      :issues-high="23"
      :licenses-high="2"
      :issues-medium="6"
      :licenses-medium="1"
      :issues-low="1"
      :licenses-low="0"
      size="small"
      compact
      :show-critical-severity="showCriticalSeverity"
      />
    </div>
    <div style="margin-bottom: 24px;">
    <BaseSeverity
    :issues-critical="3"
    :issues-high="4"
    :licenses-high="0"
    :issues-medium="2"
    :licenses-medium="1"
    :issues-low="8"
    :licenses-low="0"
    compact
    verbose
    :show-critical-severity="showCriticalSeverity"
  />
    </div>
    <div>
    <BaseSeverity style="margin-right: 24px;"
    :issues-critical="44257"
    :issues-high="18765"
    :issues-medium="237654"
    :issues-low="6784923"
    :show-critical-severity="showCriticalSeverity"
    />
    <BaseSeverity
    :issues-critical="42257"
    :issues-high="18765"
    :issues-medium="237654"
    :issues-low="6784923"
    size="small"
    :show-critical-severity="showCriticalSeverity"
    />
    </div>
  </div>
  `,
});
Critical.storyName = 'â€¦with critical';
