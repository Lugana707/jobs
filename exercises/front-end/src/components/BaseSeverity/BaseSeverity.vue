<template>
  <ul :class="containerComputedClasses" class="vue--severity">
    <template v-for="(item, index) in issueCounters">
      <li
        v-if="show(item, 'vulnerabilities')"
        :key="`${item.severity}_issue_${index}`"
        :class="countComputedClasses(item, 'vulnerabilities')"
        class="vue--severity__item"
        :data-snyk-ignore-wcag2aa="ignoreWcag2aa(item)"
      >
        <div v-if="!compact" class="vue--severity__count">
          <span
            class="vue--severity__label"
            :data-snyk-test="`Base Severity: ${item.severity} count`"
            >{{ issuesCount(item) }}</span
          >
        </div>
        <abbr
          :title="item.label + ' severity issues'"
          class="vue--severity__text"
        >
          <VulnIcon
            v-if="verbose"
            :size="14"
            class="vue--severity__type-icon"
          />
          <span class="vue--severity__label">{{ item.shortLabel }}</span>
        </abbr>
      </li>
      <li
        v-if="verbose && show(item, 'licenses')"
        :key="item.severity + '_license_' + index"
        :class="countComputedClasses(item, 'licenses')"
        class="vue--severity__item"
        :data-snyk-ignore-wcag2aa="ignoreWcag2aa(item)"
      >
        <div v-if="!compact" class="vue--severity__count">
          <span class="vue--severity__label">{{ item.licenses.count }}</span>
        </div>
        <abbr
          :title="item.label + ' severity license issues'"
          class="vue--severity__text"
        >
          <LicenseIcon
            v-if="verbose"
            :size="14"
            class="vue--severity__type-icon"
          />
          <span class="vue--severity__label">{{ item.shortLabel }}</span>
        </abbr>
      </li>
    </template>
  </ul>
</template>

<script>
import VulnIcon from '~/components/CustomIcons/Vuln';
import LicenseIcon from '~/components/CustomIcons/License';

export default {
  name: 'BaseSeverity',

  status: 'ready',

  components: {
    VulnIcon,
    LicenseIcon,
  },

  props: {
    /**
     * Number of critical severity vulnerabilities.
     */
    issuesCritical: {
      type: Number,
      default: 0,
    },
    /**
     * Number of high severity vulnerabilities.
     */
    issuesHigh: {
      type: Number,
      default: 0,
    },
    /**
     * Number of high severity license issues.
     */
    licensesHigh: {
      type: Number,
      default: 0,
    },
    /**
     * Number of medium severity vulnerabilities.
     */
    issuesMedium: {
      type: Number,
      default: 0,
    },
    /**
     * Number of medium severity license issues.
     */
    licensesMedium: {
      type: Number,
      default: 0,
    },
    /**
     * Number of low severity vulnerabilities.
     */
    issuesLow: {
      type: Number,
      default: 0,
    },
    /**
     * Number of low severity license issues.
     */
    licensesLow: {
      type: Number,
      default: 0,
    },
    /**
     * Render in compact view. Only severity labels with issues are displayed.
     */
    compact: {
      type: Boolean,
      default: false,
    },
    /**
     * Display issues broken down into vulnerabilities and license issues.
     */
    verbose: {
      type: Boolean,
      default: false,
    },
    /**
     * Adjust the size of the component
     */
    size: {
      type: String,
      default: 'default',
      validators(value) {
        return ['default', 'small'].includes(value);
      },
    },
    /**
     * If org\group supports critical severity
     * CRITICAL-SEVERITY-TAG: REMOVE WHEN CRITICAL IS OUT
     */
    showCriticalSeverity: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    issueCounters() {
      return [
        {
          severity: 'critical',
          label: 'Critical',
          shortLabel: 'C',
          vulnerabilities: {
            count: this.issuesCritical,
          },
          licenses: {
            count: 0,
          },
          total: this.issuesCritical,
        },
        {
          severity: 'high',
          label: 'High',
          shortLabel: 'H',
          vulnerabilities: {
            count: this.issuesHigh,
          },
          licenses: {
            count: this.licensesHigh,
          },
          total: this.issuesHigh + this.licensesHigh,
        },
        {
          severity: 'medium',
          label: 'Medium',
          shortLabel: 'M',
          vulnerabilities: {
            count: this.issuesMedium,
          },
          licenses: {
            count: this.licensesMedium,
          },
          total: this.issuesMedium + this.licensesMedium,
        },
        {
          severity: 'low',
          label: 'Low',
          shortLabel: 'L',
          vulnerabilities: {
            count: this.issuesLow,
          },
          licenses: {
            count: this.licensesLow,
          },
          total: this.issuesLow + this.licensesLow,
        },
      ];
    },
    containerComputedClasses() {
      return {
        'vue--severity--s': this.size === 'small',
        'vue--severity--compact': !this.verbose && this.compact,
        'vue--severity--verbose-compact': this.verbose && this.compact,
      };
    },
  },

  methods: {
    countComputedClasses(item, type) {
      return {
        [`vue--severity__item--${item.severity}`]: item[type].count > 0,
        'vue--severity__item--none': item[type].count === 0,
      };
    },
    issuesCount(item) {
      return this.formatNumber(
        this.verbose ? item.vulnerabilities.count : item.total,
      );
    },
    show(item, type) {
      if (
        type === 'vulnerabilities' &&
        this.compact &&
        this.issuesCount(item) === 0
      ) {
        return false;
      }
      if (type === 'licenses' && this.compact && item[type].count === 0) {
        return false;
      }
      if (item.severity === 'critical' && !this.showCriticalSeverity) {
        return false;
      }

      return true;
    },
    ignoreWcag2aa(item) {
      // Design decision - medium severity must be orange (not brown) and there's
      // more wiggle room for incidental information
      return item.severity === 'medium';
    },
    formatNumber(number) {
      if (number < 1000) {
        return Number.parseInt(number, 10);
      }
      if (number < 10000) {
        return Number.parseFloat(number / 1000).toFixed(1) + 'k';
      }
      if (number < 1000000) {
        return Number.parseInt(number / 1000, 10) + 'k';
      }
      return Number.parseInt(number / 1000000, 10) + 'M';
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/utils';

.vue--severity {
  $self: &;

  display: inline-flex;
  font-size: rem(14px);
  line-height: rem(24px);
  list-style: none;
  margin: 0;
  padding: 0;

  &__item {
    display: flex;
    padding: 0;
    margin: 0 space(xxs) 0 0;

    &:last-of-type {
      margin-right: 0;
    }

    &--none {
      #{$self}__text {
        background-color: color(severity, none);
      }

      #{$self}__count {
        background-color: color(severity, none-muted);
        color: color(severity, none-text);
      }
    }

    &--critical {
      #{$self}__text {
        background-color: color(severity, critical);
      }

      #{$self}__count {
        color: color(severity, critical-text);
        background-color: color(severity, critical-muted);
      }
    }

    &--high {
      #{$self}__text {
        background-color: color(severity, high);
      }

      #{$self}__count {
        color: color(severity, high-text);
        background-color: color(severity, high-muted);
      }
    }

    &--medium {
      #{$self}__text {
        background-color: color(severity, medium);
      }

      #{$self}__count {
        color: color(severity, medium-text);
        background-color: color(severity, medium-muted);
      }
    }

    &--low {
      #{$self}__text {
        background-color: color(severity, low);
      }

      #{$self}__count {
        color: color(severity, low-text);
        background-color: color(severity, low-muted);
      }
    }
  }

  &__count {
    display: inline-block;
    background-color: color(ui, white);
    font-variant: proportional-nums;
    text-align: center;
    border-right: 1px solid color(ui, white);
    width: 2rem;
    border-radius: global(border-radius, micro) 0 0 global(border-radius, micro);
  }

  // Adjusting font-variant to proportional-nums
  // causes the count to be misaligned to the
  // baseline, this span adjusts it.
  &__label {
    position: relative;
    top: 1px;
  }

  &__text {
    color: color(base, inverted);
    padding: 0 space(xs) - 2px;
    text-align: center;
    text-decoration: none;
    width: rem(24px);
    border-radius: 0 global(border-radius, micro) global(border-radius, micro) 0;
    @include font-stack(semibold);
  }

  &--s {
    font-size: rem(12px);
    line-height: rem(21px);

    #{$self}__count {
      width: 1.6rem;
    }

    #{$self}__text {
      padding: 0 space(xxs) - 2px;
      width: rem(21px);
    }

    &#{$self}--compact {
      #{$self}__text {
        border-radius: global(border-radius, micro);
        min-width: auto;
        width: rem(21px);
      }
    }
  }

  &--compact {
    #{$self}__item {
      padding-left: 0;
      padding-right: 0;
    }
    #{$self}__text {
      border-radius: global(border-radius, micro);
      padding: 0;
      text-align: center;
    }
  }

  &--verbose-compact {
    #{$self}__item {
      padding-left: 0;
      padding-right: 0;
    }
    #{$self}__text {
      border-radius: global(border-radius, micro);
      padding: 0;
      text-align: center;
      width: rem(40px);
    }
  }
}
</style>
