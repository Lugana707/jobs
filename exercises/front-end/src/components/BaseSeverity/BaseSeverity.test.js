import { shallowMount } from '@vue/test-utils';
import BaseSeverity from '~/components/BaseSeverity/BaseSeverity';

describe('BaseSeverity', () => {
  const wrapper = shallowMount(BaseSeverity);

  it('renders as expected', async () => {
    await wrapper.setProps({
      issuesCritical: 1,
      issuesHigh: 4,
      licensesHigh: 2,
      issuesMedium: 0,
      licensesMedium: 1,
      issuesLow: 8,
      licensesLow: 0,
      verbose: true,
      showCriticalSeverity: false,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with critical as expected', async () => {
    await wrapper.setProps({
      issuesCritical: 1,
      issuesHigh: 4,
      licensesHigh: 2,
      issuesMedium: 0,
      licensesMedium: 1,
      issuesLow: 8,
      licensesLow: 0,
      verbose: true,
      showCriticalSeverity: true,
    });
    expect(wrapper).toMatchSnapshot();
  });
  describe('Defaults', () => {
    const wrapper = shallowMount(BaseSeverity);

    it('defaults to zero high issues', () => {
      expect(wrapper.props('issuesHigh')).toEqual(0);
    });

    it('defaults to zero high license issues', () => {
      expect(wrapper.props('licensesHigh')).toEqual(0);
    });

    it('defaults to zero medium issues', () => {
      expect(wrapper.props('issuesMedium')).toEqual(0);
    });

    it('defaults to zero medium license issues', () => {
      expect(wrapper.props('licensesMedium')).toEqual(0);
    });

    it('defaults to zero low issues', () => {
      expect(wrapper.props('issuesLow')).toEqual(0);
    });

    it('defaults to zero low license issues', () => {
      expect(wrapper.props('licensesLow')).toEqual(0);
    });

    it('defaults to expanded', () => {
      expect(wrapper.props('compact')).toEqual(false);
    });

    it('defaults to basic', () => {
      expect(wrapper.props('verbose')).toEqual(false);
    });
  });

  describe('Computed Properties', () => {
    const wrapper = shallowMount(BaseSeverity);

    it('computes issue counters', async () => {
      await wrapper.setProps({
        issuesHigh: 3,
        licensesHigh: 4,
        issuesMedium: 2,
        licensesMedium: 1,
        issuesLow: 6,
        licensesLow: 2,
      });

      expect(wrapper.vm.issueCounters).toEqual([
        {
          severity: 'critical',
          label: 'Critical',
          shortLabel: 'C',
          vulnerabilities: {
            count: 0,
          },
          licenses: {
            count: 0,
          },
          total: 0,
        },
        {
          severity: 'high',
          label: 'High',
          shortLabel: 'H',
          vulnerabilities: {
            count: 3,
          },
          licenses: {
            count: 4,
          },
          total: 7,
        },
        {
          severity: 'medium',
          label: 'Medium',
          shortLabel: 'M',
          vulnerabilities: {
            count: 2,
          },
          licenses: {
            count: 1,
          },
          total: 3,
        },
        {
          severity: 'low',
          label: 'Low',
          shortLabel: 'L',
          vulnerabilities: {
            count: 6,
          },
          licenses: {
            count: 2,
          },
          total: 8,
        },
      ]);
    });

    it('computes container computed classes', async () => {
      const wrapper = shallowMount(BaseSeverity, {
        propsData: {
          size: 'small',
          compact: true,
        },
      });

      expect(wrapper.vm.containerComputedClasses).toEqual({
        'vue--severity--s': true,
        'vue--severity--compact': true,
        'vue--severity--verbose-compact': false,
      });

      await wrapper.setProps({ verbose: true });

      expect(wrapper.vm.containerComputedClasses).toEqual({
        'vue--severity--s': true,
        'vue--severity--compact': false,
        'vue--severity--verbose-compact': true,
      });
    });
  });

  describe('Methods', () => {
    const wrapper = shallowMount(BaseSeverity);

    it('computes classes based on params', () => {
      const item = {
        severity: 'high',
        issue: {
          count: 4,
        },
      };
      expect(wrapper.vm.countComputedClasses(item, 'issue')).toEqual({
        'vue--severity__item--high': true,
        'vue--severity__item--none': false,
      });
    });

    it('returns issue count based on verbose state', async () => {
      const item = {
        severity: 'high',
        vulnerabilities: {
          count: 4,
        },
        licenses: {
          count: 4,
        },
        total: 8,
      };
      expect(wrapper.vm.issuesCount(item)).toEqual(8);

      await wrapper.setProps({
        verbose: true,
      });
      expect(wrapper.vm.issuesCount(item)).toEqual(4);
    });

    it('returns show true only for items with issue counts in compact state', async () => {
      await wrapper.setProps({
        verbose: false,
        compact: true,
      });

      const item1 = {
        severity: 'high',
        vulnerabilities: {
          count: 4,
        },
        licenses: {
          count: 4,
        },
        total: 8,
      };
      const item2 = {
        severity: 'high',
        vulnerabilities: {
          count: 0,
        },
        licenses: {
          count: 0,
        },
        total: 0,
      };
      expect(wrapper.vm.show(item1, 'vulnerabilities')).toBeTruthy();
      expect(wrapper.vm.show(item2, 'vulnerabilities')).toBeFalsy();
      expect(wrapper.vm.show(item1, 'licenses')).toBeTruthy();
      expect(wrapper.vm.show(item2, 'licenses')).toBeFalsy();
    });

    it('truncates numbers for very high values', () => {
      expect(wrapper.vm.formatNumber(12345)).toEqual('12k');
      expect(wrapper.vm.formatNumber(1234567)).toEqual('1M');
    });
  });
});
