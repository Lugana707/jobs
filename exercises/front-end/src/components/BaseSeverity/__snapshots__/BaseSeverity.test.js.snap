// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSeverity renders as expected 1`] = `
<ul class="vue--severity">
  <!---->
  <!---->
  <li class="vue--severity__item vue--severity__item--high">
    <div class="vue--severity__count"><span data-snyk-test="Base Severity: high count" class="vue--severity__label">4</span></div> <abbr title="High severity issues" class="vue--severity__text">
      <vulnicon-stub title="Vuln Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></vulnicon-stub> <span class="vue--severity__label">H</span>
    </abbr>
  </li>
  <li class="vue--severity__item vue--severity__item--high">
    <div class="vue--severity__count"><span class="vue--severity__label">2</span></div> <abbr title="High severity license issues" class="vue--severity__text">
      <licenseicon-stub title="License Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></licenseicon-stub> <span class="vue--severity__label">H</span>
    </abbr>
  </li>
  <li data-snyk-ignore-wcag2aa="true" class="vue--severity__item vue--severity__item--none">
    <div class="vue--severity__count"><span data-snyk-test="Base Severity: medium count" class="vue--severity__label">0</span></div> <abbr title="Medium severity issues" class="vue--severity__text">
      <vulnicon-stub title="Vuln Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></vulnicon-stub> <span class="vue--severity__label">M</span>
    </abbr>
  </li>
  <li data-snyk-ignore-wcag2aa="true" class="vue--severity__item vue--severity__item--medium">
    <div class="vue--severity__count"><span class="vue--severity__label">1</span></div> <abbr title="Medium severity license issues" class="vue--severity__text">
      <licenseicon-stub title="License Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></licenseicon-stub> <span class="vue--severity__label">M</span>
    </abbr>
  </li>
  <li class="vue--severity__item vue--severity__item--low">
    <div class="vue--severity__count"><span data-snyk-test="Base Severity: low count" class="vue--severity__label">8</span></div> <abbr title="Low severity issues" class="vue--severity__text">
      <vulnicon-stub title="Vuln Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></vulnicon-stub> <span class="vue--severity__label">L</span>
    </abbr>
  </li>
  <li class="vue--severity__item vue--severity__item--none">
    <div class="vue--severity__count"><span class="vue--severity__label">0</span></div> <abbr title="Low severity license issues" class="vue--severity__text">
      <licenseicon-stub title="License Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></licenseicon-stub> <span class="vue--severity__label">L</span>
    </abbr>
  </li>
</ul>
`;

exports[`BaseSeverity renders with critical as expected 1`] = `
<ul class="vue--severity">
  <li class="vue--severity__item vue--severity__item--critical">
    <div class="vue--severity__count"><span data-snyk-test="Base Severity: critical count" class="vue--severity__label">1</span></div> <abbr title="Critical severity issues" class="vue--severity__text">
      <vulnicon-stub title="Vuln Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></vulnicon-stub> <span class="vue--severity__label">C</span>
    </abbr>
  </li>
  <li class="vue--severity__item vue--severity__item--none">
    <div class="vue--severity__count"><span class="vue--severity__label">0</span></div> <abbr title="Critical severity license issues" class="vue--severity__text">
      <licenseicon-stub title="License Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></licenseicon-stub> <span class="vue--severity__label">C</span>
    </abbr>
  </li>
  <li class="vue--severity__item vue--severity__item--high">
    <div class="vue--severity__count"><span data-snyk-test="Base Severity: high count" class="vue--severity__label">4</span></div> <abbr title="High severity issues" class="vue--severity__text">
      <vulnicon-stub title="Vuln Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></vulnicon-stub> <span class="vue--severity__label">H</span>
    </abbr>
  </li>
  <li class="vue--severity__item vue--severity__item--high">
    <div class="vue--severity__count"><span class="vue--severity__label">2</span></div> <abbr title="High severity license issues" class="vue--severity__text">
      <licenseicon-stub title="License Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></licenseicon-stub> <span class="vue--severity__label">H</span>
    </abbr>
  </li>
  <li data-snyk-ignore-wcag2aa="true" class="vue--severity__item vue--severity__item--none">
    <div class="vue--severity__count"><span data-snyk-test="Base Severity: medium count" class="vue--severity__label">0</span></div> <abbr title="Medium severity issues" class="vue--severity__text">
      <vulnicon-stub title="Vuln Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></vulnicon-stub> <span class="vue--severity__label">M</span>
    </abbr>
  </li>
  <li data-snyk-ignore-wcag2aa="true" class="vue--severity__item vue--severity__item--medium">
    <div class="vue--severity__count"><span class="vue--severity__label">1</span></div> <abbr title="Medium severity license issues" class="vue--severity__text">
      <licenseicon-stub title="License Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></licenseicon-stub> <span class="vue--severity__label">M</span>
    </abbr>
  </li>
  <li class="vue--severity__item vue--severity__item--low">
    <div class="vue--severity__count"><span data-snyk-test="Base Severity: low count" class="vue--severity__label">8</span></div> <abbr title="Low severity issues" class="vue--severity__text">
      <vulnicon-stub title="Vuln Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></vulnicon-stub> <span class="vue--severity__label">L</span>
    </abbr>
  </li>
  <li class="vue--severity__item vue--severity__item--none">
    <div class="vue--severity__count"><span class="vue--severity__label">0</span></div> <abbr title="Low severity license issues" class="vue--severity__text">
      <licenseicon-stub title="License Icon" fillcolor="currentColor" size="14" class="vue--severity__type-icon"></licenseicon-stub> <span class="vue--severity__label">L</span>
    </abbr>
  </li>
</ul>
`;
